zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/callback/ZkAsyncCallbacks.java

```diff
@@ -36,6 +36,7 @@
 public class ZkAsyncCallbacks {
   private static Logger LOG = LoggerFactory.getLogger(ZkAsyncCallbacks.class);
   public static final int UNKNOWN_RET_CODE = 255;
+  public static final int ZK_SESSION_MISMATCHED_CODE = 127;
```

Comment:
Is 127 an arbitrary number?

Reply:
Yes, sort of. I noticed that KeeperException.Code int values are all negative, so I chose a positive number (inspired by the choice of 255 for UNKNOWN_RET_CODE) to avoid any unwanted collision with those. I don't what the best practice is for choosing return codes, please let me know if I should change it to a more meaningful value.

Reply:
`ZkAsyncCallbacks` is a `zookeeper-api` module's code, so essentially it doesn't really matter and we don't have a set convention for what value we need to use.

I'd suggest that we add a block of comment here describing the situation where we should throw this exception code? For whoever gets to debug this using the code or for future developers so they know whether using this code would be appropriate.

Reply:
Also, it might be a good idea to check https://zookeeper.apache.org/doc/r3.6.2/apidocs/zookeeper-server/constant-values.html to avoid collision.

Reply:
Thanks for the resources. I'll make sure there's no collision with these constant values, and will also add appropriate comments to the code.